nil pointer when running lighthouse not specifying the env variable `GIT_SERVER` or `GIT_KIND`

```
./bin/lighthouse
{"level":"warning","msg":"no configAgent configuration","time":"2020-08-17T13:28:05+02:00"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x3b0 pc=0x1f54a9e]

goroutine 1 [running]:
github.com/jenkins-x/lighthouse/pkg/util.GetGitServer(0xc0005a9cc0, 0xc00007c280, 0x0)
	/Users/hardy/work/go/src/github.com/jenkins-x/lighthouse/pkg/util/scmclient.go:63 +0xbe
github.com/jenkins-x/lighthouse/pkg/webhook.(*Options).Run(0xc000598000, 0x0, 0x0)
	/Users/hardy/work/go/src/github.com/jenkins-x/lighthouse/pkg/webhook/webhook.go:95 +0x107
github.com/jenkins-x/lighthouse/pkg/webhook.NewCmdWebhook.func1(0xc0001ee2c0, 0x33141e8, 0x0, 0x0)
	/Users/hardy/work/go/src/github.com/jenkins-x/lighthouse/pkg/webhook/webhook.go:63 +0x2a
github.com/spf13/cobra.(*Command).execute(0xc0001ee2c0, 0xc00011a1b0, 0x0, 0x0, 0xc0001ee2c0, 0xc00011a1b0)
	/Users/hardy/work/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0xc0001ee2c0, 0xc00011c180, 0x24435b2, 0x7)
	/Users/hardy/work/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/hardy/work/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
	/Users/hardy/work/go/src/github.com/jenkins-x/lighthouse/cmd/webhooks/main.go:17 +0xd4
```

When the environment variables _GIT_SERVER_ and _GIT_KIND_ are set the controller starts, if one of them is not set, the above panic occurs. Funny enough in debug mode the code works. Seems to be a race condition of sorts!?

--- a/pkg/util/scmclient.go
+++ b/pkg/util/scmclient.go
@@ -60,8 +60,9 @@ func defaultScmTransport(scmClient *scm.Client) {
 func GetGitServer(cfg config.Getter) string {
 	serverURL := os.Getenv("GIT_SERVER")
 
-	if serverURL == "" && cfg != nil && cfg().ProviderConfig != nil {
-		serverURL = cfg().ProviderConfig.Server
+	actualConfig := cfg()
+	if serverURL == "" && actualConfig != nil && actualConfig.ProviderConfig != nil {
+		serverURL = actualConfig.ProviderConfig.Server
 	}
 	if serverURL == "" {
 		serverURL = "https://github.com"
@@ -99,8 +100,9 @@ func GetSCMClient(owner string, cfg config.Getter) (scmprovider.SCMClient, *scm.
 // GitKind gets the git kind from the environment
 func GitKind(cfg config.Getter) string {
 	kind := os.Getenv("GIT_KIND")
-	if kind == "" && cfg != nil && cfg().ProviderConfig != nil {
-		kind = cfg().ProviderConfig.Kind
+	actualConfig := cfg()
+	if kind == "" && actualConfig != nil && actualConfig.ProviderConfig != nil {
+		kind = actualConfig.ProviderConfig.Kind
 	}
 	if kind == "" {
 		kind = "github"
@@ -118,8 +120,9 @@ func GetBotName(cfg config.Getter) string {
 		}
 	}
 	botName := os.Getenv("GIT_USER")
-	if botName == "" && cfg != nil && cfg().ProviderConfig != nil {
-		botName = cfg().ProviderConfig.BotUser
+	actualConfig := cfg()
+	if botName == "" && actualConfig != nil && actualConfig.ProviderConfig != nil {
+		botName = actualConfig.ProviderConfig.BotUser
 	}
 	if botName == "" {
 		botName = "jenkins-x-bot"

